src/libraries/Directory.Build.targets

```diff
@@ -233,4 +233,19 @@
     <None Include="$(MSBuildProjectDirectory)\**\*.cs"
           Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);@(Compile)" />
   </ItemGroup>
+
+  <!--Instructs compiler not to emit .locals init, using SkipLocalsInitAttribute.-->
+  <Choose>
+    <When Condition="'$(SkipLocalsInit)' == 'true' and '$(IsNETCoreApp)' == 'true' and '$(TargetFramework)' == '$(NetCoreAppCurrent)'">
```

Comment:
It feels like this condition should just be `Condition="'$(SkipLocalsInit)' == 'true'"`, and then we default `$(SkipLocalsInit)` to `true` or `false` elsewhere based on the rest of the condition.

That way, if/when a project wants to explicitly opt in or out of this behavior, it can just override a single property.